platforms/jvm/plugins-application/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt

```diff
@@ -254,7 +254,7 @@ set -- \\
 <% } %>        -classpath "\$CLASSPATH" \\
 <% if ( mainClassName.startsWith('--module ') ) {
      %>        --module-path "\$MODULE_PATH" \\
-<% } %>        ${mainClassName} \\
+<% } %>        ${mainClassName || entryPointArgs} \\
```

Comment:
Here and in other places in the script templates where we refer to `mainClassName`, there needs to be a wrapper update in order to remove the references. We use the current templates in our build to generate the distribution scripts. A follow-up PR will take care of this.